lib/stac-browser/index.ts

```diff
@@ -0,0 +1,113 @@
+import { Stack, aws_s3 as s3, aws_s3_deployment as s3_deployment} from "aws-cdk-lib";
+import { RemovalPolicy, CfnOutput } from "aws-cdk-lib";
+import { PolicyStatement, ServicePrincipal, Effect } from "aws-cdk-lib/aws-iam";
+
+import { Construct } from "constructs";
+import { execSync } from "child_process";
+import * as fs from 'fs';
+
+export class StacBrowser extends Construct {
+
+    public bucket: s3.Bucket;
+    public bucketDeployment: s3_deployment.BucketDeployment;
+
+    constructor(scope: Construct, id: string, props: StacBrowserProps) {
+        super(scope, id);
+
+        const buildPath = this.buildApp(props.stacCatalogUrl, props.githubRepoTag);
+
+        this.bucket = new s3.Bucket(this, 'Bucket', {
+            accessControl: s3.BucketAccessControl.PRIVATE,
+            removalPolicy: RemovalPolicy.DESTROY,
+            })
+
+        this.bucket.addToResourcePolicy(new PolicyStatement({
+                    sid: 'AllowCloudFrontServicePrincipal',
+                    effect: Effect.ALLOW, 
+                    actions: ['s3:GetObject'],
+                    principals: [new ServicePrincipal('cloudfront.amazonaws.com')],
+                    resources: [this.bucket.arnForObjects('*')],
+                    conditions: {
+                        'StringEquals': {
+                            'aws:SourceArn': props.cloudFrontDistributionArn
+                        }
+                    }
+                }));
+        
+
+        this.bucketDeployment = new s3_deployment.BucketDeployment(this, 'BucketDeployment', {
```

Comment:
Could I choose an existing bucket?

Reply:
Added an option to use an existing bucket, and added a note that `cloudFrontDistributionArn` has no effect in that case, and that the identity doing the deployment should have appropriate permissions.